docs/source/user_guide.rst

```diff
@@ -0,0 +1,606 @@
+    print(response.answer)
+
+Also see :ref:`llminterface`.
+
+
+Configuring the prompt
+========================
+
+Prompt are managed through `PromptTemplate` classes. More
+specifically, the `GraphRAG` pipeline uses a `RagTemplate` with
+a default prompt. You can use another prompt by subclassing
+the `RagTemplate` class and passing it to the `GraphRAG` pipeline
+object during initialization:
+
+.. code:: python
+
+    from neo4j_genai.generation import RagTemplate, GraphRAG
+
+    # retriever = ...
+    # llm = ...
+
+    prompt_template = RagTemplate(
+        prompt="Answer the question {question} using context {context} and examples {examples}",
+        expected_inputs=["context", "question", "examples"]
+    )
+
+    rag = GraphRAG(retriever=retriever, llm=llm, prompt_template=prompt_template)
+
+    # ...
+
+
+Also see :ref:`prompttemplate`.
+
+
+The last piece you can configure in the GraphRAG pipeline is the retriever. The different options
+are described below.
+
+
+************************
+Retriever Configuration
+************************
+
+We provide implementation for the most commonly used retrievers:
+
+1. Vector retriever: performs a similarity search based on a Neo4j vector index and a query text or vector. Returns the matched node.
+2. Vector Cypher retriever: performs a similarity search based on a Neo4j vector index and a query text or vector. Returned results can be configured through a retrieval query parameter that will be executed after the index search. It can be used to fetch more context around the matched node.
+3. External retrievers: use these retrievers when your vectors are not saved in Neo4j but in an external vector database. We currently support Weaviate and Pinecone vector databases.
+4. Hybrid and Hybrid cypher retrievers: these retrievers use both a vector and full-text indexes.
+
+Retrievers all expose a `search` method that we will discuss in the next sections.
+
+Also see :ref:`retrievers`.
+
+Vector retriever
+===================
+
+The easiest way to instantiate a vector retriever is:
+
+.. code:: python
+
+    from neo4j_genai.retrievers import VectorRetriever
+
+    retriever = VectorRetriever(
+        driver,
+        index_name=POSTER_INDEX_NAME,
+    )
+
+The `index_name` is the name of the Neo4j vector index that will be used for similarity search.
+
+
+Search similar vector
+-----------------------------
+
+To find the top 3 most similar nodes, you can perform a search by vector:
+
+.. code:: python
+
+    vector = []  # a list of floats, same size as the vectors in the Neo4j vector index
+    retriever_result = retriever.search(query_vector=vector, top_k=3)
+
+However, most of the time, you will be given a text (from user) and not a vector. This
+use case is covered thanks to the `Embedder`.
+
+Search similar text
+-----------------------------
+
+When searching for a text, you must tell the retriever how to transform (embbed) the text
```

Comment:
When searching for a text, you must tell the retriever how to transform (embed) the text
```suggestion
When searching for a text, you must tell the retriever how to transform (embed) the text
```